### ISSUE TITLE
Determine form fields

### ISSUE THREAD

**Issue by kristinashu** [mozilla/network-pulse#236]
Possible fields:
- Title (Char limit = OO)
- Creators: Creators, contributors, and partners. (Might want to split people & institutions)
- URL
- Description: What would you like to share? Keep it simple and use plain language.
- Interest: Why might this be interesting to other people?
- Get involved (description): Looking for support? Describe how people can do that.
- Get involved URL: How can people learn more about the project or contact the creators?
- Thumbnail
- Featured (Boolean flag)

Not displayed:
- Network connection what would this be? (Short statement on relationship to Mozilla. Grant funded? Partner org? We don't currently collect this during submission, but could still be a useful field for tracking history in a staff dashboard.)
- Origin (Similar to above. More about the impetus driving a project. Not sure this is truly useful on frontend or backend.)
- Type (Category? (is this managed or a fluid list like user-submitted tags))

Not currently collected
- Submitted by: name, Email
- Flagged inappropriate / Pending review

Not displayed:
- Tags: Keywords to help with search by program, event, campaign, subject, etc (comma separated). Ex: mozfest, games
- The 5 Mozilla Issues (select 0,1+): Check key Issues if they relate to your thing/asset, it’s ok if it doesn’t.

**Comment by kristinashu** [comment]
@xmatthewx @acabunoc I've added some suggestions to the current fields https://docs.google.com/a/mozilla.com/document/d/1JroE86P4_BxHh7DPMp5_cLaHQ-AWgdi1BAZqS3v2m-4/edit?usp=sharing

**Comment by kristinashu** [comment]
Thanks for the feedback! 

@alanmoo @Pomax here are the fields we would like to capture in the upload form: https://docs.google.com/a/mozilla.com/document/d/1JroE86P4_BxHh7DPMp5_cLaHQ-AWgdi1BAZqS3v2m-4/edit?usp=sharing

It is similar to what we have currently and all changes have been called out in the doc. Let us know if you have any questions or concerns.

**Comment by xmatthewx** [comment]
Thanks @krisrinashu. 

A note to devs: we currently capture contributors as a single string. We will now want to capture one or more discreet entries. This will enable us later to build a list view by creator.

For UI input, we will likely have an `Add Another Creator` button, rather than requesting comma separated list in a single input. Unlike simple tags, this would leave too much room for interpretation, and yield poor data.

**Comment by alanmoo** [comment]
@xmatthewx what data should we be capturing about contributors? Name + email and/or social media?

**Comment by xmatthewx** [comment]
Creator should just be a string for name. We can complicate that later. :)

I believe we said we'd have a creators table, so we could create tag-like links and views later. Submitter should include email, but I don't think you're talking about that.

**Comment by alanmoo** [comment]
Yeah, I'm just trying to think about how we'd identify the same creator across entries since we can't string validate a name (plus names aren't unique)

**Comment by xmatthewx** [comment]
In 2018, we could check things against the CRM. In mid 2017, maybe offer autocomplete to avoid typos. For now, we'll have to be more fluid and not worry too much about potentially messy edges. All the John Smiths can celebrate their collective strength.

**Comment by Pomax** [comment]
I suggest we do as much now as we can, because updating a live database is always way more tricky than planning for the future up front - we can add an optional foreign key field to the "Creator" model that links to an existing user if there is one, which we can then in the future very easily link up once we want that link established with a single SQL query.

**Comment by xmatthewx** [comment]
Plan for it, yes. And act on those plans in ways that are easy (which I'm pretty sure is what you're suggesting). 

Skip any heavy lifts. There are many possible futures, but leveraging a creators table later depends on many other non-trivial things happening first. If our platform succeeds, a big task later can be less costly and consequential than smaller tasks now.

**Comment by alanmoo** [comment]
Based on the above, I'm working with [this ERD](https://www.lucidchart.com/documents/edit/149bc109-398b-4555-9ebd-ed350fef6445/2) to start. Any thoughts on that?

**Comment by xmatthewx** [comment]
This is great. Thanks.

I updated char limits. Based on use. In red on your diagram. 

My only question is on **submitted_by**. There will be instances when a network member submits a thing (whether on this platform or other means), and then it's moderated/entered by a staff member with auth. Amira/Julia are likely to post a ton things on behalf of others. It'd be good to capture the network member name like we do with creators. In the future we could celebrate them as a curator and connector. If that makes sense, maybe we want an entry for submitted_by separate from something like published_by. 

I realize this is atypical. Err. I'm unsure about this. Might be better served with a culture of shoutouts "e.g. via awesomeKathy" in the description. I'm curious what y'all think.

**Comment by alanmoo** [comment]
That makes sense, I had been looking at it from another standpoint (form submission automatically applying some sort of "anonymous" submitted by, but this makes more sense from a database standpoint.  @Pomax thoughts?

**Comment by xmatthewx** [comment]
@alanmoo – can we track some data on items? That would allow us to display a fav count and show popular stuff. 

MVP would be total count of views. Favs +/-. Share intent. Visit link. Contribute link. Related discussion here: https://github.com/mozilla/network-pulse/issues/69

Beyond MVP, for each data point, it would be nice to keep date info. That way we could query trending items, not just all time popular. Top activity for last 30 days or year is likely more useful than all time activity. Someone smart has probably figured out an efficient way to store and retrieve this. That person is not me.

**Comment by alanmoo** [comment]
Are we talking about Google Analytics or something baked into the API?

**Comment by xmatthewx** [comment]
Baked 🍰. Continued thread in https://github.com/mozilla/network-pulse/issues/251

**Comment by kristinashu** [comment]
Working on design in https://github.com/mozilla/network-pulse/issues/257
